build.sh

```diff
@@ -64,179 +20,4 @@ if [ -z "$HOME" ] || [ ! -d "$HOME" ]; then
-        ;;
-
-    x86_64)
-        __BuildArch=x64
-        __HostArch=x64
-        ;;
-
-    armv7l)
-        echo "Unsupported CPU $CPUName detected, build might not succeed!"
-        __BuildArch=arm
-        __HostArch=arm
-        ;;
-
-    aarch64)
-        __BuildArch=arm64
-        __HostArch=arm64
-        ;;
-
-    *)
-        echo "Unknown CPU $CPUName detected, configuring as if for x64"
-        __BuildArch=x64
-        __HostArch=x64
-        ;;
-esac
-
-# Use uname to determine what the OS is.
-OSName=$(uname -s)
-case $OSName in
-    Linux)
-        __BuildOS=Linux
-        __HostOS=Linux
-        ;;
-
-    Darwin)
-        __BuildOS=OSX
-        __HostOS=OSX
-        ;;
-
-    FreeBSD)
-        __BuildOS=FreeBSD
-        __HostOS=FreeBSD
-        ;;
-
-    OpenBSD)
-        __BuildOS=OpenBSD
-        __HostOS=OpenBSD
-        ;;
-
-    NetBSD)
-        __BuildOS=NetBSD
-        __HostOS=NetBSD
-        ;;
-
-    SunOS)
-        __BuildOS=SunOS
-        __HostOS=SunOS
-        ;;
-
-    *)
-        echo "Unsupported OS $OSName detected, configuring as if for Linux"
-        __BuildOS=Linux
-        __HostOS=Linux
-        ;;
-esac
-
-__BuildType=Debug
-__BuildArch=x64
-__SkipTests=false
-__DisableCrossgen=false
-__VerboseBuild=0
-__CrossBuild=0
-__PortableBuild=0
-
-while :; do
-    if [ $# -le 0 ]; then
-        break
-    fi
-
-    lowerI="$(echo $1 | awk '{print tolower($0)}')"
-    case $lowerI in
-        -\?|-h|--help)
-            usage
-            exit 1
-            ;;
-
-        x86)
-            __BuildArch=x86
-            ;;
-
-        x64)
-            __BuildArch=x64
-            ;;
-
-        arm)
-            __BuildArch=arm
-            ;;
-
-        armel)
-            __BuildArch=armel
-            ;;
-
-        arm64)
-            __BuildArch=arm64
-            ;;
-
-        debug)
-            __BuildType=Debug
-            ;;
-
-        release)
-            __BuildType=Release
-            ;;
-
-        cross)
-            __CrossBuild=1
-            ;;
-            
-        -portable)
-            __PortableBuild=1
-            ;;
-
-        verbose)
-            __VerboseBuild=1
-            ;;
-
-        skiptests)
-            __SkipTests=true
-            ;;
-        disablecrossgen)
-            __DisableCrossgen=true
-            ;;
-    esac
-
-    shift
-done
-
-# init the host distro name
-initHostDistroRid
-
-# init the target distro name
-initTargetDistroRid
-
-__RunArgs="-TargetArchitecture=$__BuildArch -ConfigurationGroup=$__BuildType -OSGroup=$__HostOS -DistroRid=$__DistroRid -SkipTests=$__SkipTests -DisableCrossgen=$__DisableCrossgen"
-
-if [ $__PortableBuild == 1 ]; then
-  __RunArgs="$__RunArgs -PortableBuild=True"
-fi
-
-# Configure environment if we are doing a verbose build
-if [ $__VerboseBuild == 1 ]; then
-    export VERBOSE=1
-	__RunArgs="$__RunArgs -verbose"
-fi
-
-echo "$__RunArgs"
-$DIR/run.sh build $__RunArgs
\ No newline at end of file
+$DIR/run.sh build "$@"
```

Comment:
I'm not a run/config.json expert by any means, so this could all be incorrect, but I think that section defines two things:

1) Aliases so you can write something like -Portable (or maybe it's just portable, without the dash) and have it mean -PortableBuild=True
2) Defining what parameters are always passed to MSBuild.

I think the default values are taken from the sections I showed above, and can be overwritten either by doing -ParameterName=Value or using one of the aliases.